ui/e2e_tests/datasets.dataset_name.datapoint.datapoint_id/editing.spec.ts

```diff
@@ -0,0 +1,2119 @@
+
+    // Find the newly added text editor (last contenteditable in user section)
+    const textEditor = userSection
+      .locator("div[contenteditable='true']")
+      .last();
+    await textEditor.waitFor({ state: "visible" });
+
+    // Fill with content
+    const textContent = v7();
+    await textEditor.fill(`User text content: ${textContent}`);
+
+    // Save
+    await page.getByRole("button", { name: "Save" }).click();
+    await page.waitForURL(/\/datasets\/.*\/datapoint\/[^/]+$/, {
+      timeout: 10000,
+    });
+    await page.waitForLoadState("networkidle", { timeout: 5000 });
+    await page.waitForTimeout(2000);
+
+    // Step 1: Verify text block was added
+    await expect(page.getByText(textContent)).toBeVisible();
+
+    // Step 2: Edit the text block content
+    await page.getByRole("button", { name: "Edit" }).click();
+    await expect(page.getByRole("button", { name: "Save" })).toBeVisible();
+
+    // Expand all "Show more" buttons again
+    const showMoreButton2 = page
+      .getByRole("button", { name: "Show more" })
+      .first();
+    while ((await showMoreButton2.count()) > 0) {
+      await showMoreButton2.click();
+      await page.waitForTimeout(100);
+    }
+
+    // Find the text editor again
+    const textEditor2 = userSection
+      .locator("div[contenteditable='true']")
+      .last();
+    await textEditor2.waitFor({ state: "visible" });
+
+    const textContent2 = v7();
+    await textEditor2.fill(`Edited user text: ${textContent2}`);
+
+    // Save
+    await page.getByRole("button", { name: "Save" }).click();
+    await page.waitForURL(/\/datasets\/.*\/datapoint\/[^/]+$/, {
+      timeout: 10000,
+    });
+    await page.waitForLoadState("networkidle", { timeout: 5000 });
+    await page.waitForTimeout(2000);
+
+    // Verify edited content is visible and old content is gone
+    await expect(page.getByText(textContent2)).toBeVisible();
+    await expect(page.getByText(textContent)).not.toBeVisible();
+
+    // Step 3: Delete the text block
+    await page.getByRole("button", { name: "Edit" }).click();
+    await expect(page.getByRole("button", { name: "Save" })).toBeVisible();
+
+    // Expand all "Show more" buttons again
+    const showMoreButton3 = page
+      .getByRole("button", { name: "Show more" })
+      .first();
+    while ((await showMoreButton3.count()) > 0) {
+      await showMoreButton3.click();
+      await page.waitForTimeout(100);
+    }
+
+    // Find delete button for the newly added content block (last one)
+    const deleteButton = userSection
+      .getByRole("button", { name: "Delete content block" })
+      .last();
+    await deleteButton.click();
+
+    // Save
+    await page.getByRole("button", { name: "Save" }).click();
+    await page.waitForURL(/\/datasets\/.*\/datapoint\/[^/]+$/, {
+      timeout: 10000,
+    });
+    await page.waitForLoadState("networkidle", { timeout: 5000 });
+    await page.waitForTimeout(2000);
+
+    // Verify text block was deleted
```

Comment:
Won't insist on this for the PR but these sleeps are a code smell and if we can replace with logic like this it would be better. 

// Wait for the deletion to complete by checking the element state
await page.getByText(textContent2).waitFor({ state: 'hidden', timeout: 5000 });
await page.getByText(textContent).waitFor({ state: 'hidden', timeout: 5000 });